android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -906,35 +906,42 @@ private void setUpSSL(boolean validating, HttpsURLConnection securedConnection)
 			}
 		}
 		if (sslSocketFactory == null) {
-			if (trustManagers.size() > 0 || keyManagers.size() > 0) {
-				TrustManager[] trustManagerArray = null;
-				KeyManager[] keyManagerArray = null;
-
-				if (trustManagers.size() > 0) {
-					trustManagerArray = new X509TrustManager[trustManagers.size()];
-					trustManagerArray = trustManagers.toArray(trustManagerArray);
-				}
-
-				if (keyManagers.size() > 0) {
-					keyManagerArray = new X509KeyManager[keyManagers.size()];
-					keyManagerArray = keyManagers.toArray(keyManagerArray);
-				}
-
-				try {
-					sslSocketFactory = new TiSocketFactory(keyManagerArray, trustManagerArray, tlsVersion);
-				} catch(Exception e) {
-					Log.e(TAG, "Error creating SSLSocketFactory: " + e.getMessage());
-					sslSocketFactory = null;
-				}
-			} else if (!validating) {
-				TrustManager trustManagerArray[] = new TrustManager[] { new NonValidatingTrustManager() };
-				try {
-					sslSocketFactory = new TiSocketFactory(null, trustManagerArray, tlsVersion);
-				} catch(Exception e) {
-					Log.e(TAG, "Error creating SSLSocketFactory: " + e.getMessage());
-					sslSocketFactory = null;
-				}
-			}
+		    if (trustManagers.size() > 0 || keyManagers.size() > 0) {
```

Comment:
Refactor to avoid duplicating common code?
```java
TrustManager[] trustManagerArray = null;
KeyManager[] keyManagerArray = null;
if (trustManagers.size() > 0 || keyManagers.size() > 0) {
    if (trustManagers.size() > 0) {
        trustManagerArray = new X509TrustManager[trustManagers.size()];
        trustManagerArray = trustManagers.toArray(trustManagerArray);
    }

    if (keyManagers.size() > 0) {
        keyManagerArray = new X509KeyManager[keyManagers.size()];
        keyManagerArray = keyManagers.toArray(keyManagerArray);
    }
} else if (!validating) {
    trustManagerArray = new TrustManager[] { new NonValidatingTrustManager() };
}
try {
    sslSocketFactory = new TiSocketFactory(keyManagerArray, trustManagerArray, tlsVersion);
} catch(Exception e) {
    Log.e(TAG, "Error creating SSLSocketFactory: " + e.getMessage());
    sslSocketFactory = null;
}
```